src/monty/json.py

```diff
@@ -113,7 +111,9 @@ class B(A):
         mro = type(obj).mro()
     except TypeError:
```

Comment:
I could be wrong here, I don't think this `TypeError` is related to callable type:
```python
def func():
    print("hello")

print(type(func).mro())  # >>> [<class 'function'>, <class 'object'>]
```

But more likely that we cannot call `type` on a class constructor:
```python
class A:
    def __init__(self):
        pass

print(type(A).mro())  
```

Which gives:
```
Traceback (most recent call last):
    print(type(A).mro())
          ^^^^^^^^^^^^^
TypeError: unbound method type.mro() needs an argument
```